UI: Add Launch Jumplists on Windows

WizardCM opened the pull request:
### Description

This adds basic support for Windows Jumplists. This first iteration provides the ability to choose to launch OBS with replay buffer active, minimised, or always on top.

[A future version](http://scr.wzd.li/scr/2020-05-24_18-55-54.png) could even expose Scene Collections as pinnable files, links to secondary windows like Stats or Settings, or toggles for items like Studio Mode.

![image](https://user-images.githubusercontent.com/941350/86898473-17d91780-c14c-11ea-9c2a-47f3543cfc75.png)

### Motivation and Context

Jumplists on Windows are used primarily for three things: launching an application in various states, to trigger certain screens while the app is running, or to open recent/pinned files.

OBS' launch flags/modes are not particularly discoverable. This exposes them slightly.

### How Has This Been Tested?

* Launch OBS on Windows 8 or above
* Right click the application to see the jumplists. Note that these are designed to be used when the application *isn't* running, but first launch adds them to the taskbar entry. Pin the taskbar entry to test this once OBS is launched.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

henke37 commented:
Not a bad idea in general, but I wonder if that's really the best options to offer.

notr1ch commented:
I agree with @henke37, these tasks feel very weird to show on the jump list. The main time I see jump lists is when the app is already running and on my taskbar, so having options to launch a 2nd instance makes no sense here as the user would just get a warning about running multiple instances.

I would recommend we add options that will benefit the majority of users (ie, the ones that haven't pinned OBS and are using the jump list options with OBS already running). Options like Start Streaming / Start Recording / etc would make more sense IMO.

ryantheleach commented:
I Disagree that the focus should be on already running applications. 
You say you want to benefit the majority of users, but how many existing applications make use of the jumplist, when already opened, that differ in behavior then when closed?
What are the most likely tasks that a user will reach to the taskbar for in terms of UX?

Start Streaming / Start Recording, make sense to me, but they should work regardless of the application being opened or closed.

Looking at other applications, consensus tends to be recently/frequently opened files, tabs, windows, destinations, or Tasks.

The only thing that OBS really opens is Scene Collections, and Profiles.

Possible Tasks:

Start Streaming, Start Recording, 

I like the idea of using it to help less techy users discover the launch parameters,

e.g. Open always on top, Open starting replay buffer, Open start recording, Open start streaming, Open second instance.

I wonder if it's possible to customize the Jumplist, depending on if the application is running or not, e.g. if it's streaming, have a list of scenes, or have stop streaming in the jump list.

WizardCM commented:
I've rebased on master and added the new Virtual Camera to the Jumplist.

To answer the above feedback: Jumplists are used for a variety of things depending on the application. Some require OBS to be closed, others require it to be already running. Having it provide actions like starting/stopping recording isn't a bad idea, _however_ I already have another solution for that in the works: 

![image](https://user-images.githubusercontent.com/941350/86898995-bf564a00-c14c-11ea-99a8-4084c505ec50.png)
> Start/Stop Recording/Streaming/Replay Buffer, Pause Recording & Save Replay buttons

I have almost finished the implementation of adding a thumbnail toolbar to cover the use case of "OBS is open and I want to trigger relevant actions". I just need to make better SVG icons, and then I'll be submitting a PR.

WizardCM commented:
As Microsoft plans on phasing out Jumplists in Windows 11, I'm closing this PR.
